src/Workspaces/TestAnalyzerReference/HelloWorldGenerator.cs

```diff
@@ -43,6 +44,17 @@ public static string GetMessage()
                 }
                 """, encoding: Encoding.UTF8));
 
+            context.AddSource(GeneratedEnglishClassName + "WithTime", SourceText.From($$"""
```

Comment:
any way to test balanced mode?  e.g. that when the events are fired the sg content updates (and that it doesn't if the event doesn't get fired)

Reply:
we do test balanced mode pretty heavily in tests i added.  And yes, it does exactly the sort of testing you're talking about :)